lmfit/confidence.py

```diff
@@ -293,16 +289,36 @@ def find_limit(self, para, direction):
         old_prob = 0
         limit = start_val
         i = 0
+        bound_reached = False
+        max_prob = max(self.probs)
 
-        while old_prob < max(self.probs):
+        while old_prob < max_prob:
             i = i + 1
             limit += step * direction
+            if limit > para.max:
+                limit = para.max
+                bound_reached = True
+            elif limit < para.min:
+                limit = para.min
+                bound_reached = True
 
             new_prob = self.calc_prob(para, limit)
-            rel_change = (new_prob - old_prob) / max(new_prob, old_prob, 1.e-12)
+            rel_change = (new_prob - old_prob) / max(new_prob, old_prob, 1e-12)
             old_prob = new_prob
+            if self.verbose:
+                msg = "P({}={}) = {}, max. prob={}"
+                print(msg.format(para.name, limit, new_prob, max_prob))
 
             # check for convergence
+            if bound_reached:
+                if new_prob < max(self.probs):
+                    args = para.name, limit, new_prob
+                    errmsg = ("Bound reached with "
+                              "prob({}={}) = {} < max(sigmas)"
+                              ).format(*args)
```

Comment:
I personally don't like the ```*args``` and would would prefer to use the variable names directly (as it was done previously)